lib/Dialect/FIRRTL/Transforms/LowerTypes.cpp

```diff
@@ -100,22 +100,59 @@ static bool hasZeroBitWidth(FIRRTLType type) {
       });
 }
 
-/// Return true if we can preserve the aggregate type. We can a preserve the
-/// type iff (i) the type is not passive, (ii) the type doesn't contain analog
-/// and (iii) type don't contain zero bitwidth.
-static bool isPreservableAggregateType(Type type) {
+/// Return true if the type is 1d vector.
+static bool isOneDimVectorType(FIRRTLType type) {
+  return TypeSwitch<FIRRTLType, bool>(type)
+      .Case<BundleType>([&](auto bundle) { return false; })
+      .Case<FVectorType>([&](FVectorType vector) {
+        return vector.getElementType().isGround();
+      })
+      .Default([](auto groundType) { return true; });
+}
+
+/// Return true if the type is a (possibly nested) vector type.
+static bool isVectorType(FIRRTLType type) {
+  return TypeSwitch<FIRRTLType, bool>(type)
+      .Case<BundleType>([&](auto bundle) { return false; })
+      .Case<FVectorType>([&](FVectorType vector) {
+        return isVectorType(vector.getElementType());
+      })
+      .Default([](auto groundType) { return true; });
+}
```

Comment:
It may be cleaner to make this `containsBundle` and then use the inverse of this later in the code. It was initially confusing to me that `isVectorType` is really `isVectorOrGround`.